iree/compiler/Dialect/Flow/IR/FlowDialect.cpp

```diff
@@ -60,6 +63,33 @@ Operation *FlowDialect::materializeConstant(OpBuilder &builder, Attribute value,
   return nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// Type printing and parsing
+//===----------------------------------------------------------------------===//
+
+Type FlowDialect::parseType(DialectAsmParser &parser) const {
+  Location loc = parser.getEncodedSourceLoc(parser.getNameLoc());
+  llvm::StringRef spec = parser.getFullSymbolSpec();
+  if (succeeded(parser.parseOptionalKeyword("dispatch.input"))) {
+    return DispatchInputType::parse(parser);
+  } else if (succeeded(parser.parseOptionalKeyword("dispatch.output"))) {
```

Comment:
curious why? if it's one it's never the other, so it's an if-else situation

Reply:
So I learnt that if the `if - else if -else if` chain gets too long then it breaks Windows build in the extreme. In reality this is

```
if () {

} else {
  if () {
  
  } else {
    if () {

    }
  }
}
```

AFAIK this is how it is interpreted so it causes deep nesting. Here since you are returning in every `if` you can just have `if` which has no nesting. For this case it doesnt make a difference (hence its a nit)

Reply:
As someone who exclusively builds on windows/msvc, I'll be the first to notice if an if-else causes issues :)